### PR TITLE
Let players choose team in single game mode

### PR DESCRIPTION
**Pull request by bruderol**

### CODE REVIEW

File: server/session/singleGameSession.js [hunk 1]
```diff
@@ -23,6 +23,92 @@ function createTeamsArrayForClient(session) {
...
+    if (playersInTeam === 2) {
+        // can not assign to this team, use other team.
+        teamIndex = (teamIndex === 0) ? 1 : 0;
+        playersInTeam = getPlayersInTeam(session, session.teams[teamIndex]).length;
+    }
+    let playerId = (playersInTeam * 2) + teamIndex;
+
+    // Adjust player's team name
+    let team = session.teams[teamIndex];
+    team.name = team.name + " " + playerName;
```
Comment (r3dDoX) [inline server/session/singleGameSession.js]: use template strings

File: server/session/singleGameSession.js [hunk 2]
```diff
@@ -23,6 +23,92 @@ function createTeamsArrayForClient(session) {
...
+    return session.players.filter(player => player.team.name === team.name);
+}
+
+/**
+ * @param chosenTeamIndex index of the team the player would like to join (optional, otherwise the next free place is assigned)
+ */
+function createPlayer(session, webSocket, playerName, chosenTeamIndex) {
+
+    // Calculate team and player index (depending on chosen team or assign one)
+    let teamIndex = chosenTeamIndex ? chosenTeamIndex % 2 : session.players.length % 2;
```
Comment (dstucki) [inline server/session/singleGameSession.js]: the number 0 is falsy

Comment (bruderol) [inline server/session/singleGameSession.js]: ooooh - yes - thanks for seeing this! Will correct it immediately.

File: server/session/singleGameSession.js (hunk 1 above)
Comment (bruderol) [inline server/session/singleGameSession.js]: this snippet is actually not from me (only moved it), but I agree and will improve it.

**Pull request merged**